torch/csrc/jit/attributes.h

```diff
@@ -23,62 +23,92 @@ static inline const char * toString(AttributeKind kind) {
   return names[int(kind)];
 }
 
+// Helper template for equality comparison, since at::Tensor doesn't behave like
+// a regular type.
+namespace detail {
+template <typename T>
+inline bool equal(const T& a, const T& b) {
+  return a == b;
+}
+
+template <>
+inline bool equal(const at::Tensor& a, const at::Tensor& b) {
+  return a.equal(b);
+}
+template <>
+inline bool equal(
+    const std::vector<at::Tensor>& a,
+    const std::vector<at::Tensor>& b) {
+  return a.size() == b.size() &&
+      std::equal(
+             a.cbegin(),
+             a.cend(),
+             b.cbegin(),
+             [](const at::Tensor& a, const at::Tensor& b) {
+               return a.equal(b);
+             });
+}
+
+// TODO(suo). This hack is necessary because torch.clamp uses `nan` to represent
```

Comment:
Is this safe? Not sure how people might use nan in user code. An alternative would be to just consider graphs containing nan to be unequal. This violates the regularity of Graph since a fresh copy would not equal the original, but that's not a super important property to retain.